### ISSUE TITLE
Extracting usings also grabs comments and parts of preprocessors

### ISSUE THREAD

**Issue by Emik03** [bjorndaniel/GlobalUsingsAnalyzer#7]
If you have the following code:
`Foo.cs`
```cs
// SPDX-License-Identifier: MPL-2.0
#if DEBUG
using Foo;
#endif
namespace Foo { }
```
---
Applying the lint removes only the header and includes it in the `GlobalUsings.cs` file like so:
`GlobalUsings.cs`
```cs
global // SPDX-License-Identifier: MPL-2.0
#if DEBUG
using Foo;
```
`Foo.cs`
```cs
#endif
namespace Foo { }
```
---
It preferably should be...
`GlobalUsings.cs`
```cs
global using Foo;
```
`Foo.cs`
```cs
// SPDX-License-Identifier: MPL-2.0
#if DEBUG
#endif
namespace Foo { }
```
---
Alternatively, it would be neat if preprocessors would be carried over, however I understand that it may arise other bugs or edgecases.
`GlobalUsings.cs`
```cs
#if DEBUG
global using Foo;
#endif
```
`Foo.cs`
```cs
// SPDX-License-Identifier: MPL-2.0
namespace Foo { }
```

**Comment by bjorndaniel** [comment]
Thanks for the info, I will try and fix it.
As you say moving the preprocessors might be a bit to complicated, there might be other things inside and things like that.
I will start by trying to fix so that it only moves the using and nothing else, hope that will suffice for now.

**Comment by Emik03** [comment]
That should be good enough for most use cases, I'm simply one of the few niche people who have to dabble in preprocessors frequently at the header.

**Comment by bjorndaniel** [comment]
This should now be solved in version 1.3
